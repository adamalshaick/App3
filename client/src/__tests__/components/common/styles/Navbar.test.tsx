import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import {
  StyledNavbar,
  NavigationButton,
  NavigationMarker
} from "../../../../components/common/styles/Navbar";

describe("render component", () => {
  it("renders navbar", () => {
    const component = renderer.create(<StyledNavbar />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it("renders button", () => {
    const component = renderer.create(<NavigationButton />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it("renders marker", () => {
    const Props = {
      hovered: false
    };
    const component = renderer.create(<NavigationMarker {...Props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe("button hover behaviour", () => {
  it("shows marker on button hover", () => {
    const component = renderer
      .create(<NavigationMarker hovered={true} />)
      .toJSON();
    expect(component).toHaveStyleRule("opacity", "1");
  });

  it("doesnt show marker when button isnt hovered", () => {
    const component = renderer
      .create(<NavigationMarker hovered={false} />)
      .toJSON();
    expect(component).toHaveStyleRule("opacity", "0");
  });
});
