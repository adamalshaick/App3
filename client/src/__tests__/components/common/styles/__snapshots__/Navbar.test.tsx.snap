// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders button 1`] = `
.c0 {
  background: none;
  border: none;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`render component renders marker 1`] = `
.c0 {
  padding: 0.7rem;
  font-size: 12px;
  opacity: 0.9;
  background-color: grey;
  color: white;
  border-radius: 3px;
  -webkit-transform: translate(10%,50%);
  -ms-transform: translate(10%,50%);
  transform: translate(10%,50%);
  position: absolute;
  opacity: 0;
  -webkit-transition: all 1s;
  transition: all 1s;
}

<div
  className="c0"
/>
`;

exports[`render component renders navbar 1`] = `
.c0 {
  height: 60px;
  width: 100%;
  border-bottom: whitesmoke solid 5px;
}

<header
  className="c0"
/>
`;
